### PR TITLE
Remove optional @EnableDiscoveryClient

### LINKED ISSUE #353: Remove @EnableDiscoveryClient annotations
`@EnableDiscoveryClient` annotations are now optional
See https://github.com/spring-cloud/spring-cloud-commons/issues/218

### PR DESCRIPTION
**Pull request by ryanjbaxter**
Fixes #353.

### CODE REVIEW

File: docs/src/main/asciidoc/spring-cloud-consul.adoc [hunk 1]
```diff
@@ -73,7 +72,7 @@ CAUTION: If you use <<spring-cloud-consul-config,Spring Cloud Consul Config>>, t
 
 The default service name, instance id and port, taken from the `Environment`, are `${spring.application.name}`, the Spring Context ID and `${server.port}` respectively.
 
-`@EnableDiscoveryClient` make the app into both a Consul "service" (i.e. it registers itself) and a "client" (i.e. it can query Consul to locate other services).
+To disable the auto registration of the client with Consul you can set `spring.cloud.service-registry.auto-registration.enabled` to `false`.
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-consul.adoc]: There is also a property to disable the client as well, not just registration.

Comment (ryanjbaxter) [reply docs/src/main/asciidoc/spring-cloud-consul.adoc]: Are you talking about in consul specifically or in commons?

**Comment by spencergibb** [comment]
I think there's one in both

**Comment by ryanjbaxter** [comment]
Ah is it `spring.cloud.consul.discovery.enabled`?

File: docs/src/main/asciidoc/spring-cloud-consul.adoc [hunk 2]
```diff
@@ -72,7 +72,7 @@ CAUTION: If you use <<spring-cloud-consul-config,Spring Cloud Consul Config>>, t
 
 The default service name, instance id and port, taken from the `Environment`, are `${spring.application.name}`, the Spring Context ID and `${server.port}` respectively.
 
-To disable the auto registration of the client with Consul you can set `spring.cloud.service-registry.auto-registration.enabled` to `false`.
+To disable the auto registration of the client with Consul you can set `spring.cloud.consul.discovery.enabled` to `false`.
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-consul.adoc]: auto-registration and the discovery client are two things now. I wasn't suggesting combining them.

Comment (ryanjbaxter) [reply docs/src/main/asciidoc/spring-cloud-consul.adoc]: yeah sorry, let me fix it

**Pull request merged**